Nim manual: better byref pragma explanation (#23192)
nim-lang/Nim: doc/manual.md

The `byref` pragma can be applied to an object or tuple type or a proc param.
When applied to a type it instructs the compiler to pass the type by reference
(hidden pointer) to procs. When applied to a param it will take precedence, even

> if the the type was marked as `bycopy`. When a `importc` type has `byref` pragma or

Reviewer: ```suggestion
if the the type was marked as `bycopy`. When an `importc` type has a `byref` pragma or
```

> params are marked as `byref` in `importc` proc, these params will translate to pointers.

Reviewer: ```suggestion
parameters are marked as `byref` in an `importc` proc, these params translate to pointers.
```

> When a `importcpp` type has `byref` pragma, these params will translate to

Reviewer: ```suggestion
When an `importcpp` type has a `byref` pragma, these params translate to
```

> cpp references `&`.

Reviewer: ```suggestion
C++ references `&`.
```
